### PR TITLE
DR858 - Function to process a csv file

### PR DESCRIPTION
**Pull request by JakeBlackCIC**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/integration/features/locations.feature [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@locations @pipeline_tests
+@locations @pipeline_tests @test1
```
Comment (tamaragoldschmidt) [inline tests/integration/features/locations.feature:1]: do you need this @test1 tag?

File: tests/integration/steps/data_steps.py [hunk 2]
```diff
@@ -34,3 +36,27 @@ def dynamodb_check_delete(context, id):
...
+)
+def read_csv_data(context, file_name):
+    # Open the CSV file in read mode
+    filename = read_config("csv_files", file_name)
+    file_name = os.path.basename(filename)
+    with open(filename, "r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",")  # Specify the delimiter (usually a comma)
+        headers = next(csv_reader)
+        column_count = len(headers)
+        # Convert csv_reader to a list so we can iterate over it multiple times
```
Comment (tamaragoldschmidt) [inline tests/integration/steps/data_steps.py:52]: Can there be separate steps that check for the correctness of these items?
For example: the file <insert name> has the correct data <insert value> in row <insert value> of column <insert value>
the file <insert name> has the correct row count <insert value>

File: tests/integration/features/locations.feature (hunk 1 above)
Comment (JakeBlackCIC) [reply tests/integration/features/locations.feature]: Removed this

File: tests/integration/steps/data_steps.py (hunk 2 above)
Comment (JakeBlackCIC) [reply tests/integration/steps/data_steps.py:54]: All added. Please review

**Review by tamaragoldschmidt — APPROVED** [review]

**Review by solomon209 — APPROVED** [review]

**Pull request merged**